Content.Server/Foldable/FoldableSystem.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            if (component.IsFolded)
+                args.Cancel();
+        }
+
+        private void OnFoldableInit(EntityUid uid, FoldableComponent component, ComponentInit args)
+        {
+            SetFolded(component, component.IsFolded);
+        }
+
+        private bool TryToggleFold(FoldableComponent comp)
+        {
+            return TrySetFolded(comp, !comp.IsFolded);
+        }
+
+        /// <summary>
+        /// Try to fold/unfold
+        /// </summary>
+        /// <param name="comp"></param>
+        /// <param name="state">Folded state we want</param>
+        /// <returns>True if successful</returns>
+        private bool TrySetFolded(FoldableComponent comp, bool state)
+        {
+            if (state == comp.IsFolded)
+                return false;
+
+            if (_container.IsEntityInContainer(comp.Owner))
+                return false;
+
+            // First we check if the foldable object has a strap component
+            if (EntityManager.TryGetComponent(comp.Owner, out StrapComponent? strap))
+            {
+                // If an entity is buckled to the object we can't pick it up or fold it
+                if (strap.BuckledEntities.Any())
+                    return false;
+            }
+
+            SetFolded(comp, state);
+            return true;
+        }
+
+        /// <summary>
+        /// Set the folded state of the given <see cref="FoldableComponent"/>
+        /// </summary>
+        /// <param name="component"></param>
+        /// <param name="folded">If true, the component will become folded, else unfolded</param>
+        private void SetFolded(FoldableComponent component, bool folded)
+        {
+            component.IsFolded = folded;
+            component.CanBeFolded = !_container.IsEntityInContainer(component.Owner);
+
+            // You can't buckle an entity to a folded object
+            if (EntityManager.TryGetComponent(component.Owner, out StrapComponent? strap))
+                strap.Enabled = !component.IsFolded;
+
+            // Update visuals only if the value has changed
+            if (EntityManager.TryGetComponent(component.Owner, out AppearanceComponent? appearance))
+                appearance.SetData(FoldKey, folded);
+        }
+
+        #region Event handlers
+
+        /// <summary>
+        /// Handle pickup events and block them if the object is unfolded
+        /// </summary>
+        /// <param name="uid">Target entity</param>
+        /// <param name="component">Attached foldable component</param>
+        /// <param name="args"></param>
+        private void OnInteract(EntityUid uid, FoldableComponent component, InteractHandEvent args)
+        {
+            // TODO: Storage check only exists for bodybag and coz no priority on the event.
+            if (args.Handled || EntityManager.HasComponent<EntityStorageComponent>(uid)) return;
```

Comment:
So as I understand, this is here bc body bags toggle open/closed when clicked on, and the folding interaction shouldn't conflict with that. So why not just make folding an alt-click interaction? That way it also makes it easier to un-fold things, because currently you just have to do that via the context menu.